test/conformance/memory/urMemImageGetInfo.cpp

```diff
@@ -5,93 +5,234 @@
+
+  size_t returned_width = 999;
+  ASSERT_SUCCESS(urMemImageGetInfo(image, property_name, property_size,
+                                   &returned_width, nullptr));
+
+  ASSERT_EQ(returned_width, image_desc.width);
+}
+
+TEST_P(urMemImageGetInfoTest, SuccessHeight) {
+  UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+
+  size_t property_size = 0;
+  ur_image_info_t property_name = UR_IMAGE_INFO_HEIGHT;
+
+  ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
+      urMemImageGetInfo(image, property_name, 0, nullptr, &property_size),
+      property_name);
+  ASSERT_EQ(sizeof(size_t), property_size);
+
+  size_t returned_height = 999;
+  ASSERT_SUCCESS(urMemImageGetInfo(image, property_name, property_size,
+                                   &returned_height, nullptr));
+
+  ASSERT_EQ(returned_height, image_desc.height);
+}
+
+TEST_P(urMemImageGetInfoTest, SuccessDepth) {
+  UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+
+  size_t property_size = 0;
+  ur_image_info_t property_name = UR_IMAGE_INFO_DEPTH;
+
+  ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
+      urMemImageGetInfo(image, property_name, 0, nullptr, &property_size),
+      property_name);
+  ASSERT_EQ(sizeof(size_t), property_size);
+
+  size_t returned_depth = 999;
+  ASSERT_SUCCESS(urMemImageGetInfo(image, property_name, property_size,
+                                   &returned_depth, nullptr));
+
+  ASSERT_NE(returned_depth, 999);
+}
+
+TEST_P(urMemImageGetInfoTest, SuccessArraySize) {
+  UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+
+  size_t property_size = 0;
+  ur_image_info_t property_name = UR_IMAGE_INFO_ARRAY_SIZE;
+
+  ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
+      urMemImageGetInfo(image, property_name, 0, nullptr, &property_size),
+      property_name);
+  ASSERT_EQ(sizeof(size_t), property_size);
+
+  size_t returned_depth = 999;
+  ASSERT_SUCCESS(urMemImageGetInfo(image, property_name, property_size,
+                                   &returned_depth, nullptr));
+
+  ASSERT_NE(returned_depth, 999);
+}
+
+TEST_P(urMemImageGetInfoTest, SuccessNumMipMaps) {
+  UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+
+  size_t property_size = 0;
+  ur_image_info_t property_name = UR_IMAGE_INFO_NUM_MIP_LEVELS;
+
+  ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
+      urMemImageGetInfo(image, property_name, 0, nullptr, &property_size),
+      property_name);
+  ASSERT_EQ(sizeof(size_t), property_size);
+
+  size_t returned_depth = 999;
+  ASSERT_SUCCESS(urMemImageGetInfo(image, property_name, property_size,
+                                   &returned_depth, nullptr));
+
+  ASSERT_NE(returned_depth, 999);
+}
+
+TEST_P(urMemImageGetInfoTest, SuccessNumSamples) {
+  UUR_KNOWN_FAILURE_ON(uur::LevelZero{});
+
+  size_t property_size = 0;
+  ur_image_info_t property_name = UR_IMAGE_INFO_NUM_SAMPLES;
+
+  ASSERT_SUCCESS_OR_OPTIONAL_QUERY(
+      urMemImageGetInfo(image, property_name, 0, nullptr, &property_size),
+      property_name);
+  ASSERT_EQ(sizeof(size_t), property_size);
+
+  size_t returned_depth = 999;
```

Comment:
```suggestion
  size_t returned_num_samples = 999;
```

nit: I wonder if it would just be cleaner to use an agnostic `property_value` name in all of the `urXXXGetInfoTest` test cases. Likely would improve readability and reduce risk of this slipping by.